### PR TITLE
Fix remaining `cloud.materialize.com` references.

### PR DESCRIPTION
**Pull request by antiguru**
### Motivation

Fix links that still point to `cloud.materialize.com`.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: doc/developer/platform/ux.md [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ was Snowflake's approach.
 [dbt]: https://www.getdbt.com
 [frontegg-api]: https://docs.frontegg.com/reference/getting-started-with-your-api
 [frontegg]: https://frontegg.com
-[global-api-spec]: https://cloud.materialize.com/api/schema/browse
```
Comment (antiguru) [inline doc/developer/platform/ux.md:1089]: I'm not sure what this should link to, it's broken before and after.

File: doc/developer/platform/ux.md [hunk 2]
```diff
@@ -194,7 +194,7 @@ Terraform provider to manage these resources types.
 
 ### Admin API
 
-The admin API is a REST API hosted at <https://admin.cloud.materialize.com>.
+The admin API is a REST API hosted at <https://admin.console.materialize.com>.
```
Comment (benesch) [inline doc/developer/platform/ux.md:197]: The admin API is still hosted at admin.cloud.materialize.com.

File: doc/developer/platform/ux.md [hunk 3]
```diff
@@ -209,7 +209,7 @@ either.
 
 ### Global API
 
-The global API is a REST API hosted at <https://cloud.materialize.com/api>. It
+The global API is a REST API hosted at <https://console.materialize.com/api>. It
```
Comment (benesch) [inline doc/developer/platform/ux.md:212]: Ditto, although this has moved to api.cloud.materialize.com.

File: bin/cloud-push [hunk 4]
```diff
@@ -31,6 +31,6 @@ done
 
 echo "Deploy this build to staging by running:"
 echo
-echo "    $ bin/mz --profile=staging login --endpoint=https://staging.cloud.materialize.com"
+echo "    $ bin/mz --profile=staging login --endpoint=https://staging.console.materialize.com"
```
Comment (benesch) [inline bin/cloud-push:34]: I think `mz` is a bit broken with interactive logins right now, but I think the existing `staging.cloud.materialize.com` endpoint is closer to correct than `staging.console.materialize.com`.

File: misc/python/materialize/mzcompose/services.py [hunk 5]
```diff
@@ -1039,7 +1039,7 @@ def __init__(
                         "email": username,
                         "app-password": app_password,
                         "region": region,
-                        "endpoint": f"https://{environment}.cloud.materialize.com/",
+                        "endpoint": f"https://{environment}.console.materialize.com/",
```
Comment (benesch) [inline misc/python/materialize/mzcompose/services.py:1042]: Everything from here down is correct as is (*.cloud.materialize.com), AFAICT!

File: src/mz/src/configuration.rs [hunk 6]
```diff
@@ -27,7 +27,7 @@ use crate::vault::Token;
 pub const WEB_DOCS_URL: &str = "https://www.materialize.com/docs";
 
 pub static DEFAULT_ENDPOINT: Lazy<Endpoint> =
-    Lazy::new(|| "https://cloud.materialize.com".parse().unwrap());
+    Lazy::new(|| "https://console.materialize.com".parse().unwrap());
```
Comment (joacoc) [inline src/mz/src/configuration.rs:30]: > The admin API is still hosted at admin.cloud.materialize.com.

File: src/frontegg-client/src/config.rs [hunk 7]
```diff
@@ -30,7 +30,7 @@ use crate::client::{Authentication, Client};
 /// The default endpoint the client will use to issue the requests. Currently
 /// points to Materialize admin endpoint.
 pub static DEFAULT_ENDPOINT: Lazy<Url> = Lazy::new(|| {
-    "https://admin.cloud.materialize.com"
+    "https://admin.console.materialize.com"
```
Comment (joacoc) [inline src/frontegg-client/src/config.rs:33]: > The admin API is still hosted at admin.cloud.materialize.com.

File: bin/cloud-push (hunk 4 above)
Comment (benesch) [reply bin/cloud-push:34]: @joacoc is it right that this should still be `cloud`?

Comment (benesch) [reply bin/cloud-push:34]: I think applying a quick tweak would be worth it to unblock developers. But I think this should stay as `admin.staging.cloud.materialize.com`, and then the tweak in `mz` should rewrite "cloud" to "console" when doing a browser login?

Comment (joacoc) [reply bin/cloud-push:34]: Make sense! Will prepare the PR.

Comment (antiguru) [reply bin/cloud-push:34]: Thank you!

Comment (benesch) [reply bin/cloud-push:34]: @antiguru if you back out this part of the change, I think this is good to merge?

File: doc/developer/platform/ux.md (hunk 2 above)
Comment (antiguru) [reply doc/developer/platform/ux.md]: Thanks!

File: bin/cloud-push (hunk 4 above)
Comment (antiguru) [reply bin/cloud-push]: Yup, removed.

**Comment by antiguru** [comment]
Thanks for the reviews!

**Review by benesch — APPROVED** [review]
Thanks!

**Pull request merged**